File: api/src/endpoint/runner/delete_runner.rs
use crate::endpoint::Endpoint;

pub struct DeleteRunner;

impl Endpoint for DeleteRunner {
    const PATH: &'static str = "/ci/runner/{owner}/{name}";
    const METHOD: http::Method = http::Method::DELETE;

    type Request = DeleteRunnerRequest;
    type Response = DeleteRunnerResponse;
}

pub type DeleteRunnerRequest = ();
pub type DeleteRunnerResponse = ();

File: gitdot-api-ts/src/endpoint/runner/delete-runner.ts
import { z } from "zod";

export const DeleteRunnerRequest = z.object({});
export type DeleteRunnerRequest = z.infer<typeof DeleteRunnerRequest>;

export const DeleteRunner = {
  path: "/ci/runner/{owner}/{name}",
  method: "DELETE",
  request: DeleteRunnerRequest,
  response: z.void(),
} as const;
export type DeleteRunner = typeof DeleteRunner;
